assertj-core/src/main/java/org/assertj/core/api/RecursiveComparisonAssert.java

```diff
@@ -649,6 +649,46 @@ public SELF ignoringActualEmptyOptionalFields() {
     return myself;
   }
 
+  /**
+   * Allows to ignore non-existent fields in the recursive comparison.
+   * <p>
+   * This is useful when comparing polymorphic objects where some fields might not exist in all subtypes.
+   * <p>
+   * Example:
+   * <pre><code class='java'> class BaseClass {
+   *   String commonField = "common";
+   * }
+   *
+   * class SubClass1 extends BaseClass {
+   *   String subField1 = "sub1";
+   * }
+   *
+   * class SubClass2 extends BaseClass {
+   *   // 'subField1' doesn't exist here
+   *   String subField2 = "sub2";
+   * }
+   *
+   * SubClass1 actual = new SubClass1();
+   * SubClass2 expected = new SubClass2();
+   *
+   * // Regular recursive comparison - fails due to field existence check
+   * // assertThat(actual).usingRecursiveComparison().isEqualTo(expected); // fails!
+   *
+   * // Using ignoringNonExistentFields() makes it pass
+   * assertThat(actual)
+   *   .usingRecursiveComparison()
+   *   .ignoringNonExistentFields()
+   *   .isEqualTo(expected);
+   * </code></pre>
+   *
+   * @return this {@link RecursiveComparisonAssert} to allow fluent chaining.
+   */
+  @CheckReturnValue
+  public SELF ignoringNonExistentFields() {
```

Comment:
I still need to think if we keep that name or if we should rename to `ignoringNonExistentComparedFields` which is more accurate.